src/logic/tokens/api/fetchToken.ts

```diff
@@ -1,16 +1,19 @@
 import axios from 'axios'
 
 import { getRelayUrl } from 'src/config/index'
+import { TokenDTO } from './token.dto'
 
-const fetchToken = (tokenAddress) => {
+const fetchToken = async (tokenAddress: string): Promise<TokenDTO | null> => {
   const apiUrl = getRelayUrl()
-  const url = `${apiUrl}/tokens/`
+  const url = `${apiUrl}tokens/${tokenAddress}`
 
-  return axios.get(url, {
-    params: {
-      address: tokenAddress,
-    },
-  })
+  try {
+    const result = await axios.get<TokenDTO>(url)
+    return result.data
+  } catch (error) {
+    console.error(`Fetching token data for address: ${tokenAddress} errored`, error)
```

Comment:
@lukasschor do you think is a good idea track a GA event here, when there is no token info?